src/frontend/js/components/CourseProductItem/index.tsx

```diff
@@ -11,46 +16,91 @@ const messages = defineMessages({
     description: 'Message displayed when authenticated user owned the product',
     id: 'components.CourseProductItem.enrolled',
   },
+  loading: {
+    defaultMessage: 'Loading product information...',
+    description:
+      'Accessible text for the initial loading spinner displayed when product is fetching',
+    id: 'components.CourseProductItem.loadingInitial',
+  },
 });
 
 export interface Props {
-  product: Joanie.Product;
-  order?: Joanie.OrderLite;
+  productId: Joanie.Product['id'];
+  courseCode: Joanie.Course['code'];
 }
 
-const CourseProductItem = ({ product, order }: Props) => {
-  const isOwned = order !== undefined;
+const CourseProductItem = ({ productId, courseCode }: Props) => {
+  const productQuery = useProduct(productId, { course: courseCode });
```

Comment:
Why is `course` filter needed when the `productId` is already given?

Reply:
It is totally optionnal but it ensures that the product requested is linked to the syllabus.